src/main/java/cms/gov/madie/measure/resources/MeasureController.java

```diff
@@ -121,14 +120,23 @@ public ResponseEntity<String> updateMeasure(
     Optional<Measure> persistedMeasure = repository.findById(id);
 
     if (persistedMeasure.isPresent()) {
-      if (username != null
-          && persistedMeasure.get().getCreatedBy() != null
-          && !persistedMeasure.get().isActive()) {
+      if (username != null && persistedMeasure.get().getCreatedBy() != null) {
         log.info(
             "got username [{}] vs createdBy: [{}]",
             username,
             persistedMeasure.get().getCreatedBy());
-        measureService.checkDeletionCredentials(username, persistedMeasure.get().getCreatedBy());
+        // either owner or shared-with role
+        measureService.verifyAuthorization(username, persistedMeasure.get());
+
+        // no user can update a soft-deleted measure
+        if (!persistedMeasure.get().isActive()) {
+          throw new UnauthorizedException(
+              "Measure", persistedMeasure.get().getId(), principal.getName());
+        }
```

Comment:
Should this be part of `measureService.verifyAuthorization` method?

Reply:
@adongare verifyAuthorization only verify is the user is the owner or has the shared-with role, but for a soft-deleted measure, no user (owner or shared-with or any other user) can modify the measure, so I think they are separate things.

Reply:
@adongare Thanks a lot for the suggestion Ashok! I think it makes sense. Yesterday when we talked about this, Joe mentioned: "At least until we have to concept of an Admin user, or if we elect to try out an “undo delete” feature.", so I think it's possible in the future? What do you think?

Reply:
I'm not sure either. I'm fine with leaving it as it is.